components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2048,47 +2077,77 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
     }
 
     /**
-     * Get role names according to the given domain. If the domain is not specified, roles of all the user
-     * stores will be returned.
-     *
-     * @param domainName Domain name
-     * @return Roles List
+     * Get role names according to the given domain. If the domain is not specified, roles of all the user stores
+     * will be returned.
+     * @param startIndex
+     * @param count
+     * @param domainName
+     * @return
      * @throws UserStoreException
      * @throws IdentitySCIMException
+     * @throws CharonException
      */
-    private Set<String> getRoleNamesForGroupsEndpoint(String domainName)
-            throws UserStoreException, IdentitySCIMException {
+    private Set<String> getRoleNamesForGroupsEndpoint(int startIndex, Integer count, String domainName)
+            throws UserStoreException, IdentitySCIMException, CharonException {
 
         SCIMGroupHandler groupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
+        Set<String> hybridRoles = new HashSet<>();
+        // If the domain is empty then we are retrieving all the roles and if we have the domain then reteive the
+        // roles from that domain only.
         if (StringUtils.isEmpty(domainName)) {
-            Set<String> roleNames = groupHandler.listSCIMRoles();
-            Set<String> hybridRoles = new HashSet<>(Arrays.asList(carbonUM.getHybridRoles()));
-            List<String> scimDisabledHybridRoles = getSCIMDisabledHybridRoleList(hybridRoles, roleNames);
-            if (!scimDisabledHybridRoles.isEmpty()) {
-                createSCIMAttributesForSCIMDisabledHybridRoles(scimDisabledHybridRoles);
-                roleNames.addAll(scimDisabledHybridRoles);
+            int maxGroupList = getGroupMaxLimit(domainName);
+            Set<String> roleNames;
+            // If the request is a pagination request then we are not creating the scim ids for scim disabled hybrid
```

Comment:
We should use multiline comments if the comments spans over more than a single line.